apps/minifront/src/components/shared/non-native-fee-warning.tsx

```diff
@@ -0,0 +1,78 @@
+import { BalancesResponse } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import { Metadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { getAssetIdFromValueView } from '@penumbra-zone/getters/value-view';
+import { getAssetId } from '@penumbra-zone/getters/metadata';
+import { useStakingTokenMetadata } from '../../state/shared';
+import { ReactNode } from 'react';
+
+const hasStakingToken = (
+  balancesResponses: BalancesResponse[] = [],
+  stakingAssetMetadata?: Metadata,
+): boolean => {
+  return balancesResponses.some(asset =>
+    getAssetIdFromValueView
+      .optional()(asset.balanceView)
+      ?.equals(getAssetId.optional()(stakingAssetMetadata)),
+  );
+};
+
+export const useShouldRender = (balancesResponses: BalancesResponse[] = [], amount: number) => {
+  const stakingTokenMetadata = useStakingTokenMetadata();
+  const userHasStakingToken = hasStakingToken(balancesResponses, stakingTokenMetadata.data);
+  const showNonNativeFeeWarning = amount > 0 && !userHasStakingToken;
+
+  return showNonNativeFeeWarning;
+};
+
+/**
+ * Renders a non-native fee warning if A) the user has balance of the staking
```

Comment:
A) "The user _doesn't_ have the native staking token"

Reply:
Whoops! Good catch. Fixed.